### PR TITLE
fix: clean up naming of things - `avm/res/compute/image`

### PR DESCRIPTION
**Pull request by tony-box**
## Description

- Cleaned up naming of files and parameters for better clarity

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.compute.image](https://github.com/tony-box/bicep-registry-modules/actions/workflows/avm.res.compute.image.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/compute/image/tests/e2e/max/main.wait.bicep [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+param resourceLocation string = resourceGroup().location
+
+param serviceShort string
+
+param waitTimeInSeconds string
+
+param tags object
+
+resource deployDelay 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-wait-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/max/main.wait.bicep:11]: Any chance this name could be provided via the very top `main.test.bicep` and passed down. When inspecting/troubleshooting the module you'd not only see all relevant names being set in that initial file, you'd also not need to pass top-level information like the `serviceShort`  down.

File: avm/res/compute/image/tests/e2e/waf-aligned/dependencies.bicep [hunk 2]
```diff
@@ -185,7 +194,24 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
...
+  name: '${uniqueString(deployment().name, location)}-waitForDeployment'
+  params: {
+    serviceShort: serviceShort
+    resourceLocation: location
+    waitTimeInSeconds: '15'
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/waf-aligned/dependencies.bicep:209]:
```suggestion
```

File: avm/res/compute/image/tests/e2e/waf-aligned/main.wait.bicep [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+param resourceLocation string = resourceGroup().location
+
+param serviceShort string
+
+param waitTimeInSeconds string
+
+param tags object
+
+resource deployDelay 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-wait-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/waf-aligned/main.wait.bicep:11]: same comment as for the 'max' test.

File: avm/res/compute/image/tests/e2e/max/dependencies.bicep [hunk 4]
```diff
@@ -185,7 +194,24 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
...
+  name: '${uniqueString(deployment().name, location)}-waitForDeployment'
+  params: {
+    serviceShort: serviceShort
+    resourceLocation: location
+    waitTimeInSeconds: '15'
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/max/dependencies.bicep:209]:
```suggestion
```

File: avm/res/compute/image/tests/e2e/max/dependencies.bicep [hunk 5]
```diff
@@ -185,7 +194,24 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   }
 }
 
-resource diskEncryptionSet 'Microsoft.Compute/diskEncryptionSets@2022-07-02' = {
+module waitForDeployment 'main.wait.bicep' = {
+  dependsOn: [keyVaultKeyCryptoRole]
+  scope: resourceGroup()
```
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/max/dependencies.bicep:199]: If this is running in the same scope as everything else in this template - why is in in a seperate file I wonder :)

**Comment by ReneHezser** [comment]
@tony-box 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

File: avm/res/compute/image/tests/e2e/max/dependencies.bicep (hunk 5 above)
Comment (AlexanderSehr) [reply avm/res/compute/image/tests/e2e/max/dependencies.bicep:196]: I still wonder @tony-box 😄

File: avm/res/compute/image/tests/e2e/defaults/dependencies.bicep
Comment (AlexanderSehr) [inline avm/res/compute/image/tests/e2e/defaults/dependencies.bicep:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

**Comment by jtracey93** [comment]
Hey @tony-box ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Comment by eriqua** [comment]
Hey @tony-box is this still active?

**Comment by AlexanderSehr** [comment]
Closing in favor of rebased PR #4105.
Note @tony-box, this will still list you as the author of the changes. Just rebased it to get it merged 😉

**Pull request closed without merging**